modules/services/utils/src/test/java/au/com/shiftyjelly/pocketcasts/utils/FileUtilTest.kt

```diff
@@ -0,0 +1,371 @@
+        }
+
+        assertEquals("$dir (Is a directory)", exception.message)
+    }
+
+    @Test
+    fun `fail copying file to a directory using copyFile function`() {
+        val file = tempDir.newFile()
+        val dir = tempDir.newFolder()
+
+        val exception = assertThrows(FileNotFoundException::class.java) {
+            FileUtil.copyFile(file, dir)
+        }
+
+        assertEquals("$dir (Is a directory)", exception.message)
+    }
+
+    @Test
+    fun `fail copying file to a directory using copyDirectory function`() {
+        val file = tempDir.newFile()
+        val dir = tempDir.newFolder()
+
+        val exception = assertThrows(FileNotFoundException::class.java) {
+            FileUtil.copyDirectory(file, dir)
+        }
+
+        assertEquals("$dir (Is a directory)", exception.message)
+    }
+
+    @Test
+    fun `fail copying directory to a file using copy function`() {
+        val file = tempDir.newFile()
+        val dir = tempDir.newFolder().apply {
+            File(this, "file").writeRandomBytes(100)
+        }
+
+        val exception = assertThrows(FileNotFoundException::class.java) {
+            FileUtil.copy(dir, file)
+        }
+
+        assertEquals("$file/file (Not a directory)", exception.message)
+    }
+
+    @Test
+    fun `fail copying directory to a file using copyFile function`() {
+        val file = tempDir.newFile()
+        val dir = tempDir.newFolder()
+
+        val exception = assertThrows(FileNotFoundException::class.java) {
+            FileUtil.copyFile(dir, file)
+        }
+
+        assertEquals("$dir (Is a directory)", exception.message)
+    }
+
+    @Test
+    fun `fail copying a directory to a file using copyDirectory function`() {
+        val file = tempDir.newFile()
+        val dir = tempDir.newFolder().apply {
+            File(this, "file").writeRandomBytes(100)
+        }
+
+        val exception = assertThrows(FileNotFoundException::class.java) {
+            FileUtil.copyDirectory(dir, file)
+        }
+
+        assertEquals("$file/file (Not a directory)", exception.message)
+    }
+
+    @Test
+    fun `get file name without extension for a file with a single dot`() {
+        val file = tempDir.newFile("hello.there")
+
+        val fileName = FileUtil.getFileNameWithoutExtension(file)
+
+        assertEquals("hello", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a file with multiple dots`() {
+        val file = tempDir.newFile("hello.there..traveller")
+
+        val fileName = FileUtil.getFileNameWithoutExtension(file)
+
+        assertEquals("hello.there.", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a file with no dots`() {
+        val file = tempDir.newFile("hello")
+
+        val fileName = FileUtil.getFileNameWithoutExtension(file)
+
+        assertEquals("hello", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a directory with a single dot`() {
```

Comment:
Is this test executing the same code as the `get file name without extension for a file with a single dot` test? I see the variable names are different, but I'm having trouble seeing what else is different. 

Was this maybe supposed to be `tempDir.newFolder` instead of `tempDir.newFile`? Same question for the next two tests.

Reply:
Right. That's what I get for copying and pasting. 😅